pkg/operator/operator.go

```diff
@@ -196,37 +206,45 @@ func (optr *Operator) sync(key string) error {
 }
 
 func (optr *Operator) maoConfigFromInfrastructure() (*OperatorConfig, error) {
-	infra, err := optr.osClient.ConfigV1().Infrastructures().Get("cluster", metav1.GetOptions{})
+	providerControllerImage, machineAPIOperatorImage, err := optr.imageGetterFunc()
 	if err != nil {
 		return nil, err
 	}
 
+	return &OperatorConfig{
+		TargetNamespace: optr.namespace,
+		Controllers: Controllers{
+			Provider:           providerControllerImage,
+			NodeLink:           machineAPIOperatorImage,
+			MachineHealthCheck: machineAPIOperatorImage,
+		},
+	}, nil
+}
+
+func (optr *Operator) getImages() (string, string, error) {
```

Comment:
No need to implement and mock `getImages()`. You can create `Configv1.Infrastructure` object and have the fake client return it.